app/tx_cache_bench_test.go

```diff
@@ -0,0 +1,121 @@
+package app
+
+import (
+	"crypto/rand"
+	"testing"
+)
+
+func generateRandomTxs(count int, size int) [][]byte {
+	txs := make([][]byte, count)
+	for i := range count {
+		tx := make([]byte, size)
+		_, _ = rand.Read(tx)
+		txs[i] = tx
+	}
+	return txs
+}
+
+// populateCache fills the cache with the given transactions
+func populateCache(cache *TxValidationCache, txs [][]byte) {
+	for _, tx := range txs {
+		cache.Set(tx)
+	}
+}
+
+// BenchmarkTxValidationCache_Operations benchmarks cache operations
+func BenchmarkTxValidationCache_Operations(b *testing.B) {
+	// this is the avg size of `Tx` part of `blobTx` which consists commitments from blob shares,
+	// we hash it to use as key for transaction cache,
+	// I got after checking 30000 blocks,the max was 4450
```

Comment:
this comment is good for reviewers but I guess we can delete when we want to merge:

```suggestion
```